lldb/source/Plugins/Language/Swift/SwiftHighlighter/SwiftHighlighter.swift

```diff
@@ -0,0 +1,352 @@
+import CxxStdlib
+import SwiftHighlighterSupport
+import SwiftParser
+import SwiftSyntax
+
+private extension String {
+  /// Helper function to extract ranges expressed in UTF8 offsets from a string.
+  subscript(_ range: Range<AbsolutePosition>) -> Substring.UTF8View {
+    let start = index(startIndex, offsetBy: range.lowerBound.utf8Offset)
+    let end = index(startIndex, offsetBy: range.upperBound.utf8Offset)
+    return self.utf8[start..<end]
+  }
+}
+
+private extension TriviaPiece {
+  var isComment: Bool {
+    return switch self {
+    case .lineComment, .blockComment, .docLineComment, .docBlockComment:
+      true
+    default:
+      false
+    }
+  }
+}
+
+enum AnsiColor {
+  enum Bright {
+    static let red = String(ansi.FormatAnsiTerminalCodes(std.string("${ansi.fg.bright.red}"), true))
```

Comment:
Could `String(ansi.FormatAnsiTerminalCodes(std.string(someString), true))` be a helper function?